### PR TITLE
Fix rosetta-cli tests 

### PR DESCRIPTION
**Pull request by Sventimir**
### Recent changes to the archive database schema have broken Rosetta (again).

The following problems have been fixed:
* Some ugly SQL queries are now nicely formatted (all the formatting takes place in [4eac474d04e7ede61e7da90450d3aed1aa47fa45](https://github.com/MinaProtocol/mina/commit/4eac474d04e7ede61e7da90450d3aed1aa47fa45)).
* Many Rosetta endpoints were broken completely due to SQL errors.
* Account balances were computed wrongly due to broken queries searching for relevant operations and account creation fees.
* Accounts' initial balances are now pulled from the genesis ledger instead of being assumed to be 0.

Explain how you tested your changes:
* Run the `rosetta-cli` test suite against the implementation

This is another step towards fixing Rosetta Integration tests. The final step (which will be done in a separate PR) will be fixing the scripts running these tests in the CI.

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [x] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them

* Closes [#52](https://github.com/MinaFoundation/mina/issues/52) and [#66](https://github.com/MinaFoundation/mina/issues/66) MinaFoundation repo.

### CODE REVIEW

**Comment by Sventimir** [comment]
!ci-build-me

File: src/app/rosetta/lib/account.ml [hunk 1]
```diff
@@ -282,8 +320,8 @@ module Balance = struct
       ; db_block_identifier_and_balance_info=
           (fun ~block_query ~address ->
             let (module Conn : Caqti_async.CONNECTION) = db in
-            Sql.run (module Conn) block_query address )
-      ; validate_network_choice= Network.Validate_choice.Real.validate }
+            Sql.run ~graphql_uri (module Conn) block_query address )
+      ; Validate_network_choice= Network.Validate_choice.Real.validate }
```
Comment (joaosreis) [inline src/app/rosetta/lib/account.ml:324]: Out of curiosity, why is `Validate_network_choice` with a capital V now?

Comment (Sventimir) [reply src/app/rosetta/lib/account.ml:324]: That's a good question. And even better one is why the heck does it compile!? :sweat_smile:
Fixing.

**Comment by Sventimir** [comment]
!ci-build-me

**Review by joaosreis — APPROVED** [review]

**Review by dkijania — APPROVED** [review]

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**